components/builder-worker/src/runner/workspace.rs

```diff
@@ -0,0 +1,152 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fs::File;
+use std::io::Read;
+use std::path::{Path, PathBuf};
+
+use hab_core::package::PackageArchive;
+use protocol::jobsrv as proto;
+
+use error::Result;
+
+pub struct Workspace {
+    pub job: proto::Job,
+    log: PathBuf,
+    src: PathBuf,
+    studio: PathBuf,
+    root: PathBuf,
+}
+
+impl Workspace {
+    pub fn new(data_path: String, job: proto::Job) -> Self {
+        let root = PathBuf::from(data_path).join(job.get_id().to_string());
+        Workspace {
+            job: job,
+            log: root.join("build.log"),
+            src: root.join("src"),
+            studio: root.join("studio"),
+            root: root,
+        }
+    }
+
+    pub fn last_built(&self) -> Result<PackageArchive> {
+        let build = try!(LastBuild::from_file(self.src().join("results").join("last_build.env")));
+        Ok(PackageArchive::new(self.src().join("results").join(build.pkg_artifact)))
+    }
+
+    pub fn log(&self) -> &Path {
+        &self.log
+    }
+
+    pub fn root(&self) -> &Path {
+        &self.root
+    }
+
+    pub fn src(&self) -> &Path {
+        &self.src
+    }
+
+    pub fn studio(&self) -> &Path {
+        &self.studio
+    }
+}
+
+#[derive(Debug)]
+pub struct LastBuild {
+    pub pkg_origin: String,
+    pub pkg_name: String,
+    pub pkg_version: String,
+    pub pkg_release: String,
+    pub pkg_ident: String,
+    pub pkg_artifact: String,
+    pub pkg_sha256sum: String,
+    pub pkg_blake2bsum: String,
+}
+
+impl LastBuild {
+    pub fn from_file<S: AsRef<Path>>(path: S) -> Result<Self> {
+        let mut build = LastBuild::default();
+        let mut buf: Vec<u8> = vec![];
+        let mut f = try!(File::open(path));
+        try!(f.read_to_end(&mut buf));
+        Self::parse_into(&mut build, &buf);
+        Ok(build)
+    }
+
+    pub fn parse_into(env: &mut LastBuild, buf: &[u8]) {
+        let content = String::from_utf8_lossy(buf).into_owned();
+        for line in content.lines() {
+            let split: Vec<&str> = line.split("=").map(|e| e.trim()).collect();
+            match split[0] {
+                "pkg_origin" => env.pkg_origin = split[1].to_string(),
+                "pkg_name" => env.pkg_name = split[1].to_string(),
+                "pkg_version" => env.pkg_version = split[1].to_string(),
+                "pkg_release" => env.pkg_release = split[1].to_string(),
+                "pkg_ident" => env.pkg_ident = split[1].to_string(),
+                "pkg_artifact" => env.pkg_artifact = split[1].to_string(),
+                "pkg_sha256sum" => env.pkg_sha256sum = split[1].to_string(),
+                "pkg_blake2bsum" => env.pkg_blake2bsum = split[1].to_string(),
+                field => warn!("unknown field={}", field),
```

Comment:
Before we stabilize this interface, I think we need to add target information (platform and architecture), otherwise we are going to be rev'ing it shortly.